source/core_json.c

```diff
@@ -335,7 +335,7 @@ static bool skipOneHexEscape( const char * buf,
 
     i = *start;
 #define HEX_ESCAPE_LENGTH    ( 6U )   /* e.g., \u1234 */
-    end = i + HEX_ESCAPE_LENGTH;
+    end = ( i <= ( SIZE_MAX - HEX_ESCAPE_LENGTH ) ) ? ( i + HEX_ESCAPE_LENGTH ) : SIZE_MAX;
```

Comment:
What is the rationale of using `SIZE_MAX` instead of `max` that represents the size of the buffer? Can it be using `max` instead?
`end = ( i <= ( max - HEX_ESCAPE_LENGTH ) ) ? ( i + HEX_ESCAPE_LENGTH ) : max`

Reply:
I think that would work, but I think SIZE_MAX clearly expresses the overflow avoidance and there is already a comparison to max with is symmetrical with most of the other functions.